### ISSUE TITLE
Path is not defined

### ISSUE THREAD

**Issue by ReddySk** [PrestaShop/PrestaShop#12721]
Hello, 

While automatic installation of 1.7.5.1 I am getting  stuck on 45%:
**Path not defined: supplemental/codeMappings.**

I am getting the error after I have managed workaround for problem reported here: 
https://github.com/PrestaShop/PrestaShop/issues/10161.
I have commented out the language pack downloading and it proceed to this new error. 

I am behind Proxy server with authentication. And I have suspicion ﻿that it is the root﻿ cause﻿. 
I do not see any option to setup﻿ proxy in the installation ﻿configurations.﻿ 

Apache server I am hosting on my laptop yet as a proof of concept. 
Operating System:	Microsoft Windows 10 Enterprise
Apache v 2.4.23

BTW, I can see that I have e:\Utility\XAMPP\htdocs\prestashop\translations\cldr\datas\supplemental\codeMappings.json in place. Question is if previous workaround advised by @alexmano didn't disabled the path definition somehow. 

Can you please advise how to fix that? 

﻿ Thank you, 
﻿ Cheers, Reddy

**Screenshots**
![image](https://user-images.githubusercontent.com/16384565/53420845-d889ab00-39dc-11e9-807e-c28f51f0ff4d.png)

**Additionnal information**
PrestaShop version: 1.7.5.1
PHP version: N/A

**Comment by khouloudbelguith** [comment]
Hi @ReddySk,

Thanks to create a PHP info file & check if you have some functions disabled
![image](https://user-images.githubusercontent.com/16067358/53421516-11764f80-39de-11e9-91a6-be2e5ac3becf.png)
Thanks!

**Comment by ReddySk** [comment]
Hi @khouloudbelguith 

No disabled assets yet. 
![image](https://user-images.githubusercontent.com/16384565/53422691-3e2b6680-39e0-11e9-9a63-b38ad5f37cd2.png)

Thanks

**Comment by khouloudbelguith** [comment]
@ReddySk, what is the exact workaround did you made?
What is the exact file?
Thanks!

**Comment by ReddySk** [comment]
I have edited \src\PrestaShopBundle\Install\Install.php and commented out this code at line 619-633. 

```
            //if (InstallSession::getInstance()->safe_mode) {
                EntityLanguage::checkAndAddLanguage($iso, false, true, $params_lang);
            /*} else {
                if (file_exists(_PS_TRANSLATIONS_DIR_ . (string) $iso . '.gzip') == false) {
                    $language = EntityLanguage::downloadLanguagePack($iso, _PS_INSTALL_VERSION_);

                    if ($language == false) {
                        throw new PrestashopInstallerException($this->translator->trans('Cannot download language pack "%iso%"', array('%iso%' => $iso), 'Install'));
                    }
                }

                $errors = array();
                EntityLanguage::installLanguagePack($iso, $params_lang, $errors);
            }
          */ 
```

Thanks to that it fixed the problem reported in #10161.

**Comment by khouloudbelguith** [comment]
@ReddySk, have you tried to unlock the server connection?
Could you please follow this link: https://github.com/PrestaShop/php-ps-info
phppsinfo file provides an equivalent to the phpinfo() function that reports PrestaShop Requirements information about the PHP/MySQL/Apache environment and offers suggestions for improvement.

1. Upload the file to the server. You should upload your file to the exact directory you want to test. Typically, this will be your httpdocs (/var/www/vhosts/example.com/httpdocs/) directory, although you can upload it to any subdirectory on your server as well. Use FTP to upload the file.
2. Visit the page in your browser. If you uploaded it to your html directory, you should now visit http://www.example.com/phppsinfo.php, replacing example.com with your own domain name.
3. Now you can view all of the information about PHP for your server for that particular directory.

![image](https://user-images.githubusercontent.com/16067358/44723217-a8199080-aacf-11e8-8c33-171ecbf3b5db.png)
Thanks to check and feedback.

**Comment by ReddySk** [comment]
![image](https://user-images.githubusercontent.com/16384565/53472155-3236b780-3a67-11e9-892d-1a0f67c13554.png)
![image](https://user-images.githubusercontent.com/16384565/53472174-3bc01f80-3a67-11e9-981e-86a4262b2cea.png)

**Comment by khouloudbelguith** [comment]
Hi @ReddySk,

I tried with the same workaround that you made, but I did not manage to reproduce your issue.
![image](https://user-images.githubusercontent.com/16067358/53475355-08ce5980-3a70-11e9-9f97-8ce1a5de7406.png)
It is a different issue.
It could be a server problem, thanks to check with your host & feedback.
Thanks!

**Comment by ReddySk** [comment]
Could be that problem related with the Proxy behind which I have the server?

**Comment by khouloudbelguith** [comment]
@ReddySk, We need to retrieve the PHP error log in order to find out what's wrong.
Thanks!

**Comment by ReddySk** [comment]
No error was created from PS into PHP error log. 
I have there just not related errors from another application running in parallel there. 

My error handler settings:

Directive | Local Value | Master Value
-- | -- | --
error_append_string | no value | no value
error_log | E:\Utility\XAMPP\php\logs\php_error_log | E:\Utility\XAMPP\php\logs\php_error_log
error_prepend_string | no value | no value
error_reporting | 22527 | 22527
html_errors | On | On
ignore_repeated_errors | Off | Off
log_errors | On | On
log_errors_max_len | 1024 | 1024


![image](https://user-images.githubusercontent.com/16384565/53491671-41355e00-3a97-11e9-8cb0-642d4f68bf05.png)

No record in Apache log. Only info is in access log:
```
2.252.210.142 - prestashop [27/Feb/2019:13:48:50 +0100] "GET /prestashop/install/index.php?restart=true HTTP/1.1" 200 10019 "http://some-localhost:8880/prestashop/install/index.php?restart=true" "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.99 Safari/537.36"
2.252.210.142 - prestashop [27/Feb/2019:13:48:57 +0100] "GET /prestashop/install/index.php?compile_templates=1&bo=0 HTTP/1.1" 200 - "http://some-localhost:8880/prestashop/install/index.php?restart=true" "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.99 Safari/537.36"
2.252.210.142 - prestashop [27/Feb/2019:13:48:57 +0100] "GET /prestashop/install/index.php?generateSettingsFile=true&_=1551271737049 HTTP/1.1" 200 29 "http://some-localhost:8880/prestashop/install/index.php?restart=true" "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.99 Safari/537.36"
2.252.210.142 - prestashop [27/Feb/2019:13:48:57 +0100] "GET /prestashop/install/index.php?compile_templates=1&bo=1 HTTP/1.1" 200 - "http://some-localhost:8880/prestashop/install/index.php?restart=true" "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.99 Safari/537.36"
2.252.210.142 - prestashop [27/Feb/2019:13:49:07 +0100] "GET /prestashop/install/index.php?installDatabase=true&_=1551271737050 HTTP/1.1" 200 29 "http://some-localhost:8880/prestashop/install/index.php?restart=true" "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.99 Safari/537.36"
2.252.210.142 - prestashop [27/Feb/2019:13:49:53 +0100] "GET /prestashop/install/index.php?installDefaultData=true&_=1551271737051 HTTP/1.1" 200 29 "http://some-localhost:8880/prestashop/install/index.php?restart=true" "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.99 Safari/537.36"
2.252.210.142 - prestashop [27/Feb/2019:13:50:12 +0100] "GET /prestashop/install/index.php?populateDatabase=true&_=1551271737052 HTTP/1.1" 200 29 "http://some-localhost:8880/prestashop/install/index.php?restart=true" "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.99 Safari/537.36"
2.252.210.142 - prestashop [27/Feb/2019:13:50:19 +0100] "GET /prestashop/install/index.php?configureShop=true&_=1551271737053 HTTP/1.1" 200 75 "http://some-localhost:8880/prestashop/install/index.php?restart=true" "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.99 Safari/537.36"
```

**Comment by khouloudbelguith** [comment]
@ReddySk, thanks for your feedback.
What about your server log, it could be an issue related to your server.
Thanks!

**Comment by ReddySk** [comment]
Hi @khouloudbelguith, 

I have tried to finish the installation out of office connected to internet directly and it finished with success.

It seems that there is hidden some internet check which blocks the installation in case that you have not direct connectivity to internet. 

I suppose if you will add an option for setup proxy with authentication as i.e. wordpress has, it will fix the problem for future. 

Thank you for your effort. 
You can close this case. 

Regards, Rudo

**Comment by khouloudbelguith** [comment]
@ReddySk, thanks for your feedback.
I close the issue, feel free to open a new one when needed.
Thanks!

**Comment by Igcorreia** [comment]
This is happening with the latest build. How is it possible that an online store platform has a problem like this on the install?

**Comment by khouloudbelguith** [comment]
Hi @Igcorreia,

I did not manage to reproduce the issue with PS1.7.6.7.
This issue is currently closed, can you open a new one with more information about your configuration.
Read more about how we expect the issues to be handled [here](https://build.prestashop.com/news/how-bug-reports-are-handled/). Feel free to create a new issue if you can provide further information, we will be glad to investigate it. 

Thank you!
